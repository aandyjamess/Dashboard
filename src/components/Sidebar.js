import React from "react";
import classnames from "classnames";

import { Link } from "react-router-dom";

import * as routes from "./../routes";

export default ({match: {params: {route}}}) => (
  <nav className="sidebar">
    {Object.keys(routes).map(r => (
      <Link to={r}>
        <span className={classnames("sidebar-link", {"active": route === r})}>
          <span className="sidebar-link-item icon icon-activity" />
          <span className="sidebar-link-item text-capitalize"> {r} </span>
        </span>
      </Link>
     ))}
  </nav>
);
