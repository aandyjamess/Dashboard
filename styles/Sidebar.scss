@import "./variables";

a {
 color: inherit;
 text-decoration: none;
}

.sidebar {
  position: fixed;
  left: 0px;
  top: 64px;
  bottom: 0;
  width: $nav-bar-width;
  z-index: 1000;
  background-color: $nav-default-bg;

  &-link {
    font-weight: 500;
    display: inline-block;
    width: 100%;
    padding: 15px 0;
    display: flex;
    align-items: center;
    color: grey;
    cursor: pointer;

    &-item {
      margin-left: 15px;
    }

    &:hover, &.active {
      background-color: $color-3a;
    }
  }
}

