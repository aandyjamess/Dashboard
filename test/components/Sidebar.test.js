import React from "react";
import toJson from "enzyme-to-json";

import { shallow } from "enzyme";
import { BrowserRouter } from "react-router-dom";

import Sidebar from "../../src/components/Sidebar";

describe("Sidebar", () => {
  it("renders correctly", () => {
    const match = {params: {route: "activity"}};

    const wrapper = shallow(
      <BrowserRouter>
        <Sidebar match={match} />
      </BrowserRouter>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
