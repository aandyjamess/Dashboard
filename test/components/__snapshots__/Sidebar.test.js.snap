// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Component
    match={
      Object {
        "params": Object {
          "route": "activity",
        },
      }
    }
  />
</Router>
`;
